data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -57,15 +56,14 @@ private Utils() {
         // Hide constructor, as this is a static utility class
     }
 
-    static String post(CloseableHttpClient httpClient, String url, String payload, InputStream stream, long timeoutMs,
-      Map<String, String> headers, boolean leaveOpen) throws DataServiceException, DataClientException {
-        URI uri = parseUriFromUrlString(url);
+    static String post(CloseableHttpClient httpClient, String urlStr, String payload, InputStream stream, long timeoutMs, Map<String, String> headers, boolean leaveOpen) throws DataServiceException, DataClientException {
```

Comment:
This line is way too long, break it to one line per parameter.

Reply:
Most projects by big companies have strict conventions/enforcement, including for how long a line can be. We don't, and if you think we should then we should discuss that.

I personally don't see the benefit of a strict maximum line length, and only care about readability. This personally doesn't feel too long to me - it all fits on my laptop screen, and I'd still consider it readable if just the `throws` didn't fit.

Reply:
I've used maven plugins that enforce rules like this. As above, I agree in theory that we should agree on a standard as is common in big companies, but we haven't **discussed** a standard yet and I think that should be a prerequisite to **enforcing** a standard.
(In such a discussion, I would suggest standardizing on the maximum length that fits and is easily readable on the smallest of our screens, though of course it's OSS and it's not just us reading the code. I also would be in favor of maximum parameters per line until the char limit we decide upon instead of 1 - Java already tends to be somewhat sparse.)

To convey why discussing this first would make sense, search our Java SDK with the following regex:
`.{160}`
(Mea culpa that I think all of the longest lines in the search above are mine :-) )

May I call you today to discuss standardizing?

Reply:
Sure, so I guess it shouldn't be a blocker for this PR, so I'll resolve.